src/live.asp.net/Services/YouTubeShowsService.cs

```diff
@@ -137,6 +161,32 @@ private static string GetPlaylistUrl(string playlistId)
             return $"https://www.youtube.com/playlist?list={encodedPlaylistId}";
         }
 
+        private static string ParseDuration(string duration)
+        {
+            duration = duration.ToUpper();
+
+			// youtube api format: ISO 8601 duration
+			// https://developers.google.com/youtube/v3/docs/videos#contentDetails.duration
+            // we're interested in the sequence T(#H)#M#S (e.g. PT15M33S) but a more complicate format can be returned (e.g. P3W3DT20H31M21S)
+            var time = Regex.Match(duration, @"T(\d+H)?\d+M\d+S");
```

Comment:
I used regex because the data is stored in a specific pattern, and I thought it'd be fun :smiley:!
If I may ask, why do you strongly recommend not using a regex for this?
In any case, I can rewrite this section without problems.